packages/sdk/app-framework/src/plugins/PluginHost/PluginHost.tsx

```diff
@@ -63,65 +66,100 @@ export const PluginHost = ({
+            <Root order={order} core={core} definitions={definitions} state={state.values} placeholder={placeholder} />
+          </ErrorBoundary>
+        );
       },
     },
   };
 };
 
+const DefaultFallback = ({ error }: { error: Error }) => {
+  return (
+    <div style={{ padding: '1rem' }}>
+      {/* TODO(wittjosiah): Link to docs. */}
+      <p style={{ fontSize: '0.8rem', fontStyle: 'italic' }}>
+        This is the default ErrorBoundary fallback. Consider providing your own!
+      </p>
+      <h1 style={{ fontSize: '1.2rem', fontWeight: 700, margin: '0.5rem 0' }}>{error.message}</h1>
+      <pre>{error.stack}</pre>
+    </div>
+  );
+};
+
+type RootProps = {
+  order: PluginDefinition['meta'][];
+  state: PluginContext;
+  definitions: Record<string, () => Promise<PluginDefinition>>;
+  core: string[];
+  placeholder: ReactNode;
+};
+
+const Root = ({ order, core, definitions, state, placeholder }: RootProps) => {
+  const [error, setError] = useState<unknown>();
+
+  useEffect(() => {
+    log('initializing plugins', { enabled: state.enabled });
+    const timeout = setTimeout(async () => {
+      try {
+        const enabledIds = [...core, ...state.enabled].sort((a, b) => {
+          const indexA = order.findIndex(({ id }) => id === a);
+          const indexB = order.findIndex(({ id }) => id === b);
+          return indexA - indexB;
+        });
+
+        const enabled = await Promise.all(
+          enabledIds
+            .map((id) => definitions[id])
+            // If local storage indicates a plugin is enabled, but it is not available, ignore it.
+            .filter((definition): definition is () => Promise<PluginDefinition> => Boolean(definition))
+            .map((definition) => definition()),
+        );
+
+        const plugins = await Promise.all(
+          enabled.map(async (definition) => {
+            const plugin = await initializePlugin(definition).catch((err) => {
+              console.error('Failed to initialize plugin:', definition.meta.id, err);
+              return undefined;
+            });
+            return plugin;
+          }),
+        ).then((plugins) => plugins.filter((plugin): plugin is Plugin => Boolean(plugin)));
+        log('plugins initialized', { plugins });
+
+        await Promise.all(enabled.map((pluginDefinition) => pluginDefinition.ready?.(plugins)));
+        log('plugins ready', { plugins });
+
+        state.plugins = shallow(plugins);
+        state.ready = true;
+      } catch (err) {
+        setError(err);
+      }
+    });
+
+    return () => {
+      clearTimeout(timeout);
+      state.ready = false;
+      // TODO(wittjosiah): Does this ever need to be called prior to having dynamic plugins?
+      // void Promise.all(enabled.map((definition) => definition.unload?.()));
+    };
+  }, []);
+
+  if (error) {
+    throw error;
+  }
+
+  if (!state.ready) {
+    return <>{placeholder}</>;
+  }
+
+  const ComposedContext = composeContext(state.plugins);
```

Comment:
This is an antipattern — we should avoid as much as possible defining components within the render function of a component, as it causes reactivity issues.

I can see this was present in the previous version of the code, but worth keeping an eye on this instance.